app/controllers/users/registrations_controller.rb

```diff
@@ -9,5 +16,15 @@ def update
         respond_with resource
       end
     end
+
+    private
+
+    def their_own?
+      current_user.id.eql?(params[:id].to_i)
+    end
+
+    def authorize_for_current_user
+      authorize current_user if their_own?
```

Comment:
Actually looking at this again, authorising `current_user` doesn't make sense, cause this will always be true. We need somehow run `authorize user` and the `user` has to be the one being changed. 

So something like this:

```ruby
user = User.find(params[:id])
authorize user, :edit_password?
```